WooCommerce/Classes/Authentication/Epilogue/StorePickerViewController.swift

```diff
@@ -362,13 +399,20 @@ private extension StorePickerViewController {
     /// Re-initializes the Login Flow, forcing a logout. This may be required if the WordPress.com Account has no Stores available.
     ///
     func restartAuthentication() {
-        guard ServiceLocator.stores.needsDefaultStore, let navigationController = navigationController else {
+        guard ServiceLocator.stores.needsDefaultStore else {
             return
         }
 
         ServiceLocator.stores.deauthenticate()
 
         let loginViewController = ServiceLocator.authenticationManager.loginForWordPressDotCom()
+
+        guard let navigationController = navigationController else {
+            let navController = UINavigationController(nibName: nil, bundle: nil)
+            navController.setViewControllers([loginViewController], animated: true)
```

Comment:
We're not doing anything with `navController` here. Maybe dead code.. or a missing commit?

Reply:
setViewControllers sets the root view controller to the loginViewController and pops the StorePickerVC. Unless I have that wrong? I updated it from creating a new navController, setting a rootVC to the loginVC, and then dismissing this VC.

Reply:
Inside the guard, a new UINavigationController is instantiated, `setViewControllers` is called (which is great!). But the new instance you've got in **navControlller** is lost, immediatelly right after the return in L413.

Reply:
`navController` here is a new `UINavigationController` instance, its view controllers are set but it doesn't seem to be presented or used in the navigation stack after this line. I think in this else case, we can just return (and maybe `asserttionFailure` before)?

Reply:
Yep, you are correct. This isn't where I should be trying to restore navigation. The code to check for a nil navigationController was already here, hinting at a problem that couldn't be found previously. I found where the issue was, fixed that and added an `assertFailure()` as suggested here.

To reproduce the bug, check out commit `eac673c` and try this login flow:

1. Log in to a valid WP.com account that doesn't have any WC Stores. 
2. The user kills the app and re-runs it. (Or Stop Xcode and re-run, same thing.) 
3. The app delegate restores the current state, and the Store Picker screen displays again. 

Only this time, the Store Picker is set as the root view controller, which is a problem because we want to pop to root-- which should be the login view controller.